lib/rspec/core/formatters/documentation_formatter.rb

```diff
@@ -38,6 +38,10 @@ def example_failed(failure)
           output.puts failure_output(failure.example)
         end
 
+        def message(notification)
+          output.puts "#{current_indentation}#{notification.message}"
```

Comment:
For a complete solution I think this should be:

```
@messages << notification.message
```

with a new method:
```
def output_messages
  @messages.each do |message|
    output.puts "#{current_indentation}  #{message}"
  end
  @messages = []
```

This should be called after line 25 in finished, and called in all 3 example methods after there own message.

Reply:
Yes, I thought about doing this.

1/ The log messages are buffered which means that they might never be printed if there is a critical failure. It also means for long running specs, the output is not interactive. The nice thing about documentation formatter is that it incrementally prints progress.

2/ It changes the existing behaviour of `message` significantly.

I don't really care about 2/ so much but worth considering. I'm more concerned about 1/. However, if you don't care about it, then we can make this change.

Reply:
Okay PR updated according to your requirements.

It's fine for my use case and I do think it makes sense. I think the value of having messages indented within the example group logical and nicely formatted, but the trade off of interactivity, I'm not sure about it. Well, in my case, I don't care, but it's a regression on existing behaviour, bigger than just adding some indentation.

Reply:
> 1/ The log messages are buffered which means that they might never be printed if there is a critical failure. It also means for long running specs, the output is not interactive. The nice thing about documentation formatter is that it incrementally prints progress.

It's the same as the progress, messages are printed with each example so its no "worse" than before, RSpec takes care to print its messages in the event of a critical failure, the only time it won't is if you "force quit it"

> 2/ It changes the existing behaviour of message significantly.

> I don't really care about 2/ so much but worth considering. I'm more concerned about 1/. However, if you don't care about it, then we can make this change.

I consider indenting at all changing the existing behaviour of message significantly. If we care about this we can leave it as is.

Reply:
I personally think this is a good PR, because output is generally better. However, I understand if you don't want to merge it because it affects existing behaviour. I defer to your judgement since you obviously know the space much better than me.